core/tee/tee_svc_cryp.c

```diff
@@ -3106,11 +3126,11 @@ TEE_Result syscall_cipher_init(unsigned long state, const void *iv,
 		res = crypto_cipher_init(cs->ctx, cs->mode,
 					 (uint8_t *)(key1 + 1), key1->key_size,
 					 (uint8_t *)(key2 + 1), key2->key_size,
-					 iv, iv_len);
+					 iv_bbuf, iv_len);
 	} else {
 		res = crypto_cipher_init(cs->ctx, cs->mode,
 					 (uint8_t *)(key1 + 1), key1->key_size,
-					 NULL, 0, iv, iv_len);
+					 NULL, 0, iv_bbuf, iv_len);
 	}
```

Comment:
add `bb_free(iv_bbuf, iv_len)` before line 3135.

Reply:
According to @jenswi-linaro

> There is normally no need to free bounce buffers at the end of a syscall_*() function since the next thing to happen is that the bounce buffers are reset. In fact, I believe it's better to skip freeing bounce buffers in syscall_*() functions unless they are allocated in a loop or in other ways may accumulate.
> 
> Please consider this pattern in all the other syscall functions here too.

`bb_free()`s were originally there, but I removed them. The rationale behind this afaiu is that the bounce buffer offset will be reset to 0 after returning from the syscall handler so there will be no memory leak anyway. Would you prefer to explicitly freeing them?

Reply:
Okay thanks!